### PR TITLE
Niddy Griddy

### PR DESCRIPTION
**Pull request by designatednerd**
Got the grid up and running and showing a bunch of food emoji

### CODE REVIEW

**Comment by designatednerd** [comment]
@vokal/ios-developers May I please request an actual code review? Trying to see if anyone knows swiftier ways to do some of this

**Comment by chillpop** [comment]
sure

File: SwiftSnake/SwiftSnake/SKNodes/GridBackgroundNode.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+//  Copyright © 2015 Vokal. All rights reserved.
+//
+
+import Foundation
+import SpriteKit
+
+class GridBackgroundNode: SKNode {
+    
+    //Since TVs are 16x9...
+    let columns = 16 * 2
```
Comment (chillpop) [inline SwiftSnake/SwiftSnake/SKNodes/GridBackgroundNode.swift]: private?

File: SwiftSnake/SwiftSnake/SKNodes/GridBackgroundNode.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    
+    static func backgroundNode() -> GridBackgroundNode {
+        let node = GridBackgroundNode()
+        node.name = NodeType.Background.rawValue
+        return node
+    }
+    
+    func setupGrid(gridSize: CGSize) {
+        
+        let xWidth = (gridSize.width / CGFloat(columns)) as CGFloat
```
Comment (chillpop) [inline SwiftSnake/SwiftSnake/SKNodes/GridBackgroundNode.swift]: do you need the cast?

File: SwiftSnake/SwiftSnake/SKNodes/GridBackgroundNode.swift [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        let yHeight = (gridSize.height / CGFloat(rows)) as CGFloat
+        
+        for column in 0..<columns {
+            let xOrigin = xWidth * CGFloat(column) + xWidth / 2
+            for row in 0..<rows {
+                let yOrigin = yHeight * CGFloat(row) + yHeight / 2
+                let space = SpaceNode.emptyNodeOfSize(CGSize(width: xWidth, height: yHeight))
```
Comment (chillpop) [inline SwiftSnake/SwiftSnake/SKNodes/GridBackgroundNode.swift]: is it worth pulling out this unchanging CGSize into a `let` somewhere?

File: SwiftSnake/SwiftSnake/SKNodes/NodeConstants.swift [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+//
+
+import Foundation
+
+enum NodeType: String {
+    case Food = "Food",
+    Background = "Background",
+    Space = "Space",
+    SnakeHead = "SnakeHead",
+    SnakeBody = "SnakeBody"
```
Comment (chillpop) [inline SwiftSnake/SwiftSnake/SKNodes/NodeConstants.swift]: in swift 2, you don't need to explicitly declare the enum string values (unless you want them to be different from the names)

also, add a trailing comma?

Comment (chillpop) [inline SwiftSnake/SwiftSnake/SKNodes/NodeConstants.swift]: nope, no trailing comma

File: SwiftSnake/SwiftSnake/ViewControllers/GameViewController.swift [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        // create a new scene
+        let scene = SnakeGameScene.makeSceneOfSize(skView.frame.size)
+        skView.presentScene(scene)
+        
+        // add a tap gesture recognizer
+        let tapGesture = UITapGestureRecognizer(target: self, action: "handleTap:")
+        var gestureRecognizers = [UIGestureRecognizer]()
+        gestureRecognizers.append(tapGesture)
+        if let existingGestureRecognizers = skView.gestureRecognizers {
+            gestureRecognizers.appendContentsOf(existingGestureRecognizers)
```
Comment (chillpop) [inline SwiftSnake/SwiftSnake/ViewControllers/GameViewController.swift]: does `skView.addGestureRecognizer(...)` not work?

Comment (designatednerd) [inline SwiftSnake/SwiftSnake/ViewControllers/GameViewController.swift]: I should probably nuke this from the boilerplate too - it was doing this on a scene kit view for which the answer is apparently no.

File: SwiftSnake/SwiftSnake/SKNodes/GridBackgroundNode.swift (hunk 1 above)
Comment (designatednerd) [inline SwiftSnake/SwiftSnake/SKNodes/GridBackgroundNode.swift]: yep, good call.

File: SwiftSnake/SwiftSnake/SKNodes/GridBackgroundNode.swift (hunk 2 above)
Comment (designatednerd) [inline SwiftSnake/SwiftSnake/SKNodes/GridBackgroundNode.swift]: Hm, maybe not now that I'm dividing by a cg float...

File: SwiftSnake/SwiftSnake/SKNodes/GridBackgroundNode.swift (hunk 3 above)
Comment (designatednerd) [inline SwiftSnake/SwiftSnake/SKNodes/GridBackgroundNode.swift]: I'd been doing that earlier but it was barfing - let me try it again.

File: SwiftSnake/SwiftSnake/SKNodes/NodeConstants.swift (hunk 4 above)
Comment (designatednerd) [inline SwiftSnake/SwiftSnake/SKNodes/NodeConstants.swift]: oooooooh, right! forgot about the explicit thing. eeeeeexcellent

**Comment by designatednerd** [comment]
@chillpop Changes pushed. Thanks for taking the time! Lemme know how this looks.

File: SwiftSnake/SwiftSnake/SKNodes/GridBackgroundNode.swift [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    
+    static func backgroundNode() -> GridBackgroundNode {
+        let node = GridBackgroundNode()
+        node.name = NodeType.Background.rawValue
+        return node
+    }
+    
+    func setupGrid(gridSize: CGSize) {
+        let xWidth = (gridSize.width / CGFloat(columns))
+        let yHeight = (gridSize.height / CGFloat(rows))
```
Comment (chillpop) [inline SwiftSnake/SwiftSnake/SKNodes/GridBackgroundNode.swift]: unnecessary parens, but that's ok

**Comment by chillpop** [comment]
LGTM

**Comment by designatednerd** [comment]
Parentheses nuked!

**Comment by chillpop** [comment]
:shipit:

**Pull request merged**